api/server/backend/build/backend.go

```diff
@@ -88,11 +88,9 @@ func (b *Backend) Build(ctx context.Context, config backend.BuildConfig) (string
 		}
 	}
 
-	if !useBuildKit {
+	if imageID != "" && !useBuildKit {
 		stdout := config.ProgressWriter.StdoutFormatter
 		fmt.Fprintf(stdout, "Successfully built %s\n", stringid.TruncateID(imageID))
```

Comment:
Really nit-picking; this could be a single line, and use `Fprintln`;

```suggestion
		_, _ = fmt.Fprintln(config.ProgressWriter.StdoutFormatter, "Successfully built", stringid.TruncateID(imageID))
```

Reply:
Done